### PR TITLE
Graceful handling of Numba import errors

### LINKED ISSUE #332: Handle `numba` errors more gracefully
If `numba` raises an error other than an `ImportError`, `tobac` cannot be imported. We should handle these errors more gracefully.

### PR DESCRIPTION
**Pull request by freemansw1**
Resolves #332 and handles `numba` import errors more gracefully

* [x] Have you followed our guidelines in CONTRIBUTING.md? 
* [x] Have you self-reviewed your code and corrected any misspellings? 
* [ ] Have you written documentation that is easy to understand?
* [x] Have you written descriptive commit messages? 
* [ ] Have you added NumPy docstrings for newly added functions? 
* [x] Have you formatted your code using black? 
* [ ] If you have introduced a new functionality, have you added adequate unit tests?
* [x] Have all tests passed in your local clone? 
* [ ] If you have introduced a new functionality, have you added an example notebook?
* [x] Have you kept your pull request small and limited so that it is easy to review? 
* [ ] Have the newest changes from this branch been merged?

### CODE REVIEW

File: tobac/utils/internal.py
Comment (w-k-jones) [inline tobac/utils/internal.py:446]: Should this be `raise kie` or will it raise the keyboard exception implicitly?

Comment (freemansw1) [reply tobac/utils/internal.py:446]: This is intentional; if you `raise kie`, you lose the stack trace. See also https://docs.python.org/3/tutorial/errors.html#raising-exceptions

Comment (w-k-jones) [reply tobac/utils/internal.py:446]: Ah great, I didn't know that!

**Comment by freemansw1** [comment]
I will re-request reviews after #336 is merged

**Comment by freemansw1** [comment]
@JuliaKukulies @kelcyno @w-k-jones I've resolved the merge conflict and this is now ready for review.

**Review by JuliaKukulies — APPROVED** [review]
Looks good to me!

**Comment by freemansw1** [comment]
I'm not sure why the notebooks are failing, but the action seems to be failing at the installing dependencies step.

**Comment by freemansw1** [comment]
Will wait on #340 to merge/re-review

**Review by w-k-jones — APPROVED** [review]

**Pull request merged**
